Tests/FDBTests/FDBTests.swift

```diff
@@ -123,6 +130,58 @@ class FDBTests: XCTestCase {
         let result = try fdb.get(key: updatedKey)
         XCTAssertNotNil(result)
         try XCTAssertEqual(result!.cast() as String, value)
+        
+        
+        let asyncKey = subspace[FDB.Versionstamp(userData: 42)]["myAsyncKey"]
+        let asyncValue: String = "world!"
+        
+        let versionstampedKeyValueFuture: EventLoopFuture<String?> = (fdb.withTransaction(on: self.eventLoop) { transaction in
+            return transaction
+                .set(versionstampedKey: asyncKey, value: getBytes(asyncValue.utf8))
+                .flatMap { transaction in
+                    return transaction.getVersionstamp(commit: true)
+                }
+        } as EventLoopFuture<FDB.Versionstamp>)
```

Comment:
This looks very ugly :D Try to avoid this casting. It can be done, for example, by specifying the return type at 138 and 141.

Reply:
I've done it this way (code from my sandbox):

```swift
let result: String? = try connection
    .withTransaction(on: eventLoop) { (transaction: AnyFDBTransaction) in
        try! transaction
            .set(versionstampedKey: asyncKey, value: Bytes(asyncValue.utf8))
            .flatMap { (_) -> EventLoopFuture<FDB.Versionstamp> in
                let versionstamp: EventLoopFuture<FDB.Versionstamp> = transaction.getVersionstamp()
                return transaction.commit().flatMap { versionstamp }
            }
    }
    .flatMap { (versionstamp: FDB.Versionstamp) -> EventLoopFuture<String?> in
        var versionstamp = versionstamp
        dump(versionstamp)
        versionstamp.userData = 42
        dump(versionstamp)

        return connection.withTransaction(on: eventLoop) { transaction in
            transaction
                .get(key: subspace[versionstamp]["myAsynckey"], commit: true)
                .map { (maybeBytes, _) -> String? in
                    guard let bytes = maybeBytes else {
                        return nil
                    }
                    return String(bytes: bytes, encoding: .ascii)!
                }
        }
    }
    .wait()
```